osu.Game.Tournament/IPC/FileBasedIPC.cs

```diff
@@ -146,54 +156,143 @@ public Storage LocateStableStorage()
+                    {
                         return stableInstallPath;
-                }
-                catch
-                {
+                    }
                 }
 
-                try
-                {
-                    using (RegistryKey key = Registry.ClassesRoot.OpenSubKey("osu"))
-                        stableInstallPath = key?.OpenSubKey(@"shell\open\command")?.GetValue(string.Empty).ToString().Split('"')[1].Replace("osu!.exe", "");
+                return stableInstallPath;
+            }
+            finally
+            {
+                Logger.Log($"Stable path for tourney usage: {stableInstallPath}");
+            }
+        }
 
-                    if (checkExists(stableInstallPath))
-                        return stableInstallPath;
-                }
-                catch
-                {
-                }
+        private void saveStablePath()
+        {
+            using (var stream = tournamentStorage.GetStream(stable_config, FileAccess.Write, FileMode.Create))
+            using (var sw = new StreamWriter(stream))
+            {
+                sw.Write(JsonConvert.SerializeObject(stableInfo,
+                    new JsonSerializerSettings
+                    {
+                        Formatting = Formatting.Indented,
+                        NullValueHandling = NullValueHandling.Ignore,
+                        DefaultValueHandling = DefaultValueHandling.Ignore,
+                    }));
+            }
+        }
 
-                stableInstallPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), @"osu!");
-                if (checkExists(stableInstallPath))
-                    return stableInstallPath;
+        private string findFromEnvVar()
+        {
+            try
+            {
+                Logger.Log("Trying to find stable with environment variables");
+                string stableInstallPath = Environment.GetEnvironmentVariable("OSU_STABLE_PATH");
 
-                stableInstallPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), ".osu");
                 if (checkExists(stableInstallPath))
+                {
+                    stableInfo.StablePath.Value = stableInstallPath;
+                    saveStablePath();
                     return stableInstallPath;
+                }
+            }
+            catch
+            {
+            }
 
-                return null;
+            return null;
+        }
+
+        private string findFromJsonConfig()
+        {
+            try
+            {
+                Logger.Log("Trying to find stable through the json config");
+
+                if (!string.IsNullOrEmpty(stableInfo.StablePath.Value))
+                    return stableInfo.StablePath.Value;
             }
-            finally
+            catch
             {
-                Logger.Log($"Stable path for tourney usage: {stableInstallPath}");
             }
+
+            return null;
+        }
```

Comment:
This method has an actively misleading name (it doesn't read any JSON config or find anything in one, it just reads a previously-found path).